src/smartpeak/source/ui/WindowSizesAndPositions.cpp

```diff
@@ -58,6 +58,13 @@ namespace SmartPeak
     right_window_x_pos_ = left_window_x_size_ + bottom_and_top_window_x_size_;
   }
   void WindowSizesAndPositions::setWindowSizesAndPositions(const bool& show_top_window, const bool& show_bottom_window, const bool& show_left_window, const bool& show_right_window) {
+    // reset perc for bottom if needed
+    if (!show_bottom_window_ && show_bottom_window)
```

Comment:
Is the need for `show_bottom_window_` and the extra logic a bug on our side or a bug on the IMGUI side?

Reply:
It's a bug on our side. once the bottom window is hidden, the top window takes all the space (this is our logic) and modifies the perc (space percentage that takes the bottom window), And then when we want to display the bottom window again, the perc stays to 0. with that modification, we can detect that we want to display back the bottom window and reset the perc.

Reply:
Another approach to have split windows would be to use that trick here: https://github.com/ocornut/imgui/issues/319. I think we could remove all these calculations we make by hand. It's not huge work, but maybe not so worth it as we have something working.

Reply:
That looks like a promising approach.  If we can reduce code at the same time I think it would be well worth it.